models/biocro/R/run.biocro.R

```diff
@@ -51,111 +44,177 @@ run.biocro <- function(lat, lon, metpath, soil.nc = NULL, config = config, coppi
       day1 <- NULL
       dayn <- NULL
     }
+    if (!is.null(day1) && day1 > lubridate::yday(starti)) {
```

Comment:
This if statement would be easier to understand as
lubridate::yday(starti) = max(day1, lubridate::yday(starti))

Reply:
`starti` and `endi` are used between this assignment and their initialization. It could all be in one place.
```
lubridate::yday(starti) = max(c(day1, start.date, lubridate::ymd(paste0(yeari, "-01-01")))
```

Reply:
(but using `<-` instead of `=` per PEcAn style guide)

Reply:
The tricky bit with assigning all 3 at once is that start.date could be in a different year, in which case we don't care about its DOY.  I think it's probably better to keep these as separate calculations, but have cleaned up the logic  so it's more clear how they fit together.